File: LeetCode/LeetCode 75/9_Palindrome_Number/9_Palindrome_Number.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0) {
            return false
        }
        val nums: MutableList<Int> = mutableListOf()
        
        var remain: Int = x

        while(remain > 0) {
            val last_digit = remain % 10
            remain = remain / 10
            nums.add(last_digit)
        }

        for ((i, num) in nums.withIndex()) {
            if (num != nums[nums.size-1-i]) {
                return false
            }
        }
        return true
    }
}

fun main() {
    val s = Solution()
    val x = -121
    println(s.isPalindrome(x))
}

File: LeetCode/LeetCode 75/9_Palindrome_Number/9_Palindrome_Number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        nums = []

        if x < 0:
            nums.append(999)
            x = abs(x)

        remain = x
        while(remain > 0):
            last_number = remain % 10
            remain = int(remain / 10)
            nums.append(last_number)
        
        length = len(nums)
        for i, num in enumerate(nums):
            if num != nums[length-1-i]:
                return False
        return True

x = 10

s = Solution()
print(s.isPalindrome(x))
